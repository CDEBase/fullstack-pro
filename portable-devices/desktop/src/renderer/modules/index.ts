import modules, { MainRoute} from './module';
import { logger } from '@cdm-logger/client';
import { ClientTypes } from '@common-stack/client-react';
class UtilityClass {
    // tslint:disable-next-line:no-shadowed-variable
    constructor(private modules) {
    }
    public getCacheKey(storeObj) {
        return this.modules.getDataIdFromObject(storeObj);
    }
}

const utility = new UtilityClass(modules);

// additional bindings to container
const container = modules.createContainers({}) as any;
container.bind(ClientTypes.Logger).toConstantValue(logger);
container.bind(ClientTypes.UtilityClass).toConstantValue(utility);

export default modules;
export { MainRoute, container };
