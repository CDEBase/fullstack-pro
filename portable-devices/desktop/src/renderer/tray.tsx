const { createStore, applyMiddleware } = require('redux');
const {
  forwardToMain,
  replayActionRenderer,
  getInitialStateRenderer,
  createAliasedAction,
} = require('electron-redux');

import { connectedReactRouter_counter } from '../reducers'
import { increment,decrement } from '../actions'
import { ipcRenderer } from 'electron'
import { toInteger } from 'lodash';

// setup store
const initialState = getInitialStateRenderer();
const store = createStore(connectedReactRouter_counter, initialState, applyMiddleware(forwardToMain));

replayActionRenderer(store);

// set up renderer
function mount() {
  document.getElementById('app').innerHTML = `
    <p>
      Counter: <span id="value">0</span> 
      <button id="increment">+</button>
      <button id="decrement">-</button>
      <button id="notify">Click</button>
    </p>
  `;

  document.getElementById('increment').addEventListener('click', () => {
    store.dispatch(increment());
    var current_count:String = (toInteger(document.getElementById('value').innerHTML) + 1).toString(); 
    ipcRenderer.send('update-title-tray-window-event', current_count);
  });

  document.getElementById('decrement').addEventListener('click', () => {
    store.dispatch(decrement());
    var current_count:String = (toInteger(document.getElementById('value').innerHTML) - 1).toString(); 
    ipcRenderer.send('update-title-tray-window-event', current_count);
  });

  document.getElementById('notify').addEventListener('click', () => {
      // var current_count:String = document.getElementById('value').innerHTML;
      let notif = new window.Notification( 'My First Notification', {
        body: 'Body of Notification'
      })
      notif.onclick = function(){
        ipcRenderer.send('show-about-window-event')
      }
  });
}

function renderValue() {
  document.getElementById('value').innerHTML = store.getState().toString();
}

mount();
renderValue();

store.subscribe(renderValue);

