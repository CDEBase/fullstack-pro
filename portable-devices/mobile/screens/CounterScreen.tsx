import { DrawerScreenProps } from "@react-navigation/drawer";
import * as React from "react";
import { Button, View, Text } from "react-native";
import { DrawerScreensParamList } from "../types/app-types";

function CounterScreen(
  props: DrawerScreenProps<DrawerScreensParamList, "Notifications">
) {
  const { navigation } = props;
  const [counter, setCounter] = React.useState(0);

  return (
    <View style={{ flex: 1, alignItems: "center", justifyContent: "center" }}>
      <Button
        onPress={() => navigation.goBack()}
        title="Go back home"
        testID="go-back-home-btn"
      />

      <View style={{ marginTop: 20 }}>
        <View style={{ marginBottom: 20 }}>
          <Text>Counter value: {counter}</Text>
        </View>

        <Button
          onPress={() => setCounter((c) => c + 1)}
          title="Increment Counter"
          testID="inc-btn"
        />

        <View style={{ marginTop: 10 }}>
          <Button
            onPress={() => setCounter(0)}
            title="Reset Counter"
            testID="reset-btn"
          />
        </View>
      </View>
    </View>
  );
}

export default CounterScreen;
