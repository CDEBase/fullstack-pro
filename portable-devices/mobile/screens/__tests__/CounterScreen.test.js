import * as React from "react";
import renderer from "react-test-renderer";
import CounterScreen from "../CounterScreen";
import { fireEvent, render } from "@testing-library/react-native";

test("renders correctly", () => {
  const tree = renderer.create(<CounterScreen />).toJSON();

  //   expect(tree).toMatchSnapshot();
});

test("all buttons and text are rendering correctly", () => {
  const { getAllByTestId, findByText } = render(<CounterScreen />);

  getAllByTestId("go-back-home-btn");
  findByText("Counter value:");
  findByText("Go back home");
  findByText("Increment Counter");
  findByText("Reset Counter");
});

test("increment btn working correctly", () => {
  const { getAllByTestId, findByText } = render(<CounterScreen />);

  const incBtn = getAllByTestId("inc-btn")[0];

  // before pressing increment button, counter value should be zero
  findByText(/counter value: 0/gi);
  fireEvent.keyPress(incBtn);

  // after pressing increment button, counter value should be one
  findByText(/counter value: 1/gi);
});

test("reset button working correctly", () => {
  const { findByText, getAllByTestId } = render(<CounterScreen />);
  const resetBtn = getAllByTestId("reset-btn")[0];

  fireEvent.keyPress(resetBtn);
  // after pressing the reset btn, the counter value should reset to zero
  findByText(/counter value: 0/gi);
});
