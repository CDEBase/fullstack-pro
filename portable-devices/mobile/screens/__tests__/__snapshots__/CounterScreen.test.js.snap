// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Go back home
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "marginTop": 20,
      }
    }
  >
    <View
      style={
        Object {
          "marginBottom": 20,
        }
      }
    >
      <Text>
        Counter value: 
        0
      </Text>
    </View>
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Increment Counter
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "marginTop": 10,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Reset Counter
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
